stream/src/main/java/org/apache/apex/malhar/stream/api/impl/ApexStreamImpl.java

```diff
@@ -0,0 +1,457 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.stream.api.impl;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.apex.malhar.stream.api.ApexStream;
+import org.apache.apex.malhar.stream.api.function.Function;
+import org.apache.apex.malhar.stream.api.operator.FunctionOperator;
+import org.apache.commons.beanutils.BeanUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.conf.Configuration;
+
+import com.datatorrent.api.Attribute;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DAG;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.LocalMode;
+import com.datatorrent.api.Operator;
+import com.datatorrent.lib.algo.UniqueCounter;
+import com.datatorrent.lib.io.ConsoleOutputOperator;
+import com.datatorrent.stram.plan.logical.LogicalPlan;
+
+/**
+ * Default stream implementation for ApexStream interface.
+ * It creates the dag(execution plan) from stream api
+ */
+public class ApexStreamImpl<T> implements ApexStream<T>
+{
+
+  static Set<Attribute> OPERATOR_ATTRIBUTES;
+
+  static Set<Attribute> DAG_ATTRIBUTES;
+
+  static Set<Attribute> INPUT_ATTRIBUTES;
+
+  static Set<Attribute> OUTPUT_ATTRIBUTES;
+
+  static {
+
+    OPERATOR_ATTRIBUTES = new HashSet<>();
+    DAG_ATTRIBUTES = new HashSet<>();
+    INPUT_ATTRIBUTES = new HashSet<>();
+    OUTPUT_ATTRIBUTES = new HashSet<>();
+
+    try {
+      for (Field field : Context.OperatorContext.class.getDeclaredFields()) {
+        if (field.getType() == Attribute.class) {
+          OPERATOR_ATTRIBUTES.add((Attribute)field.get(Context.OperatorContext.class));
+        }
+      }
+
+      for (Field field : Context.DAGContext.class.getDeclaredFields()) {
+        if (field.getType() == Attribute.class) {
+          DAG_ATTRIBUTES.add((Attribute)field.get(Context.DAGContext.class));
```

Comment:
@siyuanh : If I take DAG object and set non DAG level attributes there, will high level apis work?
Can you please give an example?